pkg/metadata/host/host.go

```diff
@@ -106,7 +106,7 @@ func GetPythonVersion() string {
 // getHostAliases returns the hostname aliases from different provider
 // This should include GCE, Azure, Cloud foundry, kubernetes
 func getHostAliases() []string {
-	aliases := []string{}
+	aliases := config.Datadog.GetStringSlice("host_aliases")
```

Comment:
We should add a test around this new behavior. Also do we dedup aliases somewhere ? I don't think we needed before as we would not get multiple time the same alias but we might now.

Reply:
Also, I think host aliases must respect RFC1123 like hostname does (to double check). If it's a case we need to apply the check here too a drop the invalid ones,

Reply:
> We should add a test around this new behavior.

I considered adding a test for `getHostAliases()` as there isn't any now, but the kublet provider talks to external system and isn't configurable. This means the test result would potentially depend on the environment, which I think isn't a good behavior for a unit test. Perhaps there is an obvious way to refactor this to make testing easer that I am missing?

> Also do we dedup aliases somewhere ? I don't think we needed before as we would not get multiple time the same alias but we might now.

Good point, added.

> Also, I think host aliases must respect RFC1123 like hostname does (to double check). If it's a case we need to apply the check here too a drop the invalid ones,

Looks like the back end is happy with whatever we send it (arbitrary ascii, unicode, emojis), so I'm not sure there is anything we could reject.